### ISSUE TITLE
High cpu usage in 0.0.4 release

### ISSUE THREAD

**Issue by bostrt** [bostrt/quikey#12]
pystack output below. The threads are in state below while high CPU is happening but the functionality of quikey remains fine. I think there must be an uncaught exception somewhere (?) 

```
# pystack 15563
Dumping Threads....


  File "/usr/lib64/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/__init__.py", line 136, in run
    self._run()
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/keyboard/_xorg.py", line 499, in _run
    super(Listener, self)._run()
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/xorg.py", line 394, in _run
    self._context, self._handler)
  File "/usr/lib/python3.6/site-packages/Xlib/ext/record.py", line 243, in enable_context
    context = context)
  File "/usr/lib/python3.6/site-packages/Xlib/ext/record.py", line 220, in __init__
    rq.ReplyRequest.__init__(self, *args, **keys)
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/rq.py", line 1369, in __init__
    self.reply()
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/rq.py", line 1381, in reply
    self._display.send_and_recv(request = self._serial)
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/display.py", line 613, in send_and_recv
    gotreq = self.parse_response(request)
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/display.py", line 705, in parse_response
    gotreq = self.parse_request_response(request) or gotreq
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/display.py", line 793, in parse_request_response
    req._parse_response(self.data_recv[:self.recv_packet_len])
  File "/usr/lib/python3.6/site-packages/Xlib/ext/record.py", line 224, in _parse_response
    self._callback(r)
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/__init__.py", line 154, in inner
    return f(self, *args, **kwargs)
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/xorg.py", line 458, in _handler
    self._handle(self._display_stop, event)
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/keyboard/_xorg.py", line 518, in _handle
    self.on_press(key)
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/__init__.py", line 75, in inner
    if f(*args) is False:
  File "/home/rbost/.local/lib/python3.6/site-packages/quikey/input.py", line 152, in __call__
    if self.notifier.notify(self.dequetostr()):
  File "/home/rbost/.local/lib/python3.6/site-packages/quikey/input.py", line 67, in notify
    if observer.notify(key):
  File "/home/rbost/.local/lib/python3.6/site-packages/quikey/input.py", line 25, in notify
    keyboard = Controller()
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/keyboard/_xorg.py", line 165, in __init__
    self.ALT_GR_MASK = alt_gr_mask(self._display)
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/xorg.py", line 128, in alt_gr_mask
    display.__altgr_mask = _find_mask(display, 'Mode_switch')
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/xorg.py", line 94, in _find_mask
    for index, keycodes in enumerate(display.get_modifier_mapping()):
  File "/usr/lib/python3.6/site-packages/Xlib/display.py", line 944, in get_modifier_mapping
    r = request.GetModifierMapping(display = self.display)
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/rq.py", line 1369, in __init__
    self.reply()
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/rq.py", line 1381, in reply
    self._display.send_and_recv(request = self._serial)
  File "/usr/lib/python3.6/site-packages/Xlib/protocol/display.py", line 562, in send_and_recv
    rs, ws, es = select.select([self.socket], writeset, [], timeout)

---------------

  File "/usr/lib64/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/rbost/.local/lib/python3.6/site-packages/quikey/filewatch.py", line 22, in run
    for event in self.inotify.read():
  File "/home/rbost/.local/lib/python3.6/site-packages/inotify_simple/inotify_simple.py", line 136, in read
    pending = self._poller.poll(timeout)

---------------

  File "/home/rbost/.local/bin/quikey-daemon", line 11, in <module>
    sys.exit(cli())
  File "/home/rbost/.local/lib/python3.6/site-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/home/rbost/.local/lib/python3.6/site-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/home/rbost/.local/lib/python3.6/site-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/rbost/.local/lib/python3.6/site-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/rbost/.local/lib/python3.6/site-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/home/rbost/.local/lib/python3.6/site-packages/quikey/qkdaemon.py", line 97, in start
    main(foreground, buffer_size, trigger_keys)
  File "/home/rbost/.local/lib/python3.6/site-packages/quikey/qkdaemon.py", line 76, in main
    listener.join()
  File "/home/rbost/.local/lib/python3.6/site-packages/pynput/_util/__init__.py", line 192, in join
    super(AbstractListener, self).join(*args)
  File "/usr/lib64/python3.6/threading.py", line 1056, in join
    self._wait_for_tstate_lock()
  File "/usr/lib64/python3.6/threading.py", line 1072, in _wait_for_tstate_lock
    elif lock.acquire(block, timeout):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <module>


```

**Comment by bostrt** [comment]
Should not be creating a new `Controller` each PhraseHandler method call.

**Comment by bostrt** [comment]
d55feab  seems to have resolved this. Released in 0.0.5.
